File: htdocs/submit.py
#!/usr/bin/env python3
import os
import cgi

# X-Powered-By: PHP/7.4.10
# Content-type: text/html; charset=UTF-8


# Get the content length from the environment or set it to 0
content_length = int(os.environ.get("CONTENT_LENGTH", 0))

# Print HTTP headers
print("X-Powered-By: Python/3.8.5")
print("Content-type: text/html; charset=UTF-8")
print("\n\r\n\r")

# Print the HTML page
print("<html>")
print("<head>")
print("<title>Login Page</title>")
print('<link rel="stylesheet" type="text/css" href="../styles/style.css">')
print("</head>")
print("<body>")

# Check if the request method is POST
if "REQUEST_METHOD" in os.environ and os.environ["REQUEST_METHOD"] == "POST":
    # Parse the POST data using the cgi module
    form = cgi.FieldStorage()   

    # Retrieve the value of the 'username' and 'password' fields from the form
    if "username" in form and "password" in form:
        username = form.getvalue("username")
        password = form.getvalue("password")

        # Display a welcome message
        print('<div class="welcome-container">')
        print(f"<h2>Welcome, {username}!</h2>")
        print(f"<p>Your password is {password}.</p>")
        print('</div>')
    else:
        # Display an error message
        print('<div class="error-container">')
        print("<p>Error: Username or password not provided.</p>")
        print('</div>')
else:
    # Display the login form
    print('<div class="login-form-container">')
    print('<a href="form.html">LOGIN</a>')
    print('</div>')

print("</body>")
print("</html>")

File: htdocs/submit.php
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Welcome User</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            background-color: #f4f4f4;
            margin: 0;
            padding: 0;
            display: flex;
            align-items: center;
            justify-content: center;
            height: 100vh;
        }

        .welcome-container {
            text-align: center;
            padding: 20px;
            background-color: #fff;
            border-radius: 8px;
            box-shadow: 0 0 10px rgba(0, 0, 0, 0.1);
        }

        h1 {
            color: #333;
        }
    </style>
</head>
<body>

<?php
// Check if the username is provided in the POST data
    
    if (isset($_POST['username']))
    {
        $password = $_POST['password'];
        $password = htmlspecialchars($password, ENT_QUOTES, 'UTF-8');
        if ($password == 'password')
        {
            $username = htmlspecialchars($_POST['username'], ENT_QUOTES, 'UTF-8');
            echo '<div class="welcome-container">';
            echo '<h1>Welcome, ' . $username . '!</h1>';
            echo '</div>';
        }
        else
        {
            echo '<div class="welcome-container">';
            echo '<h1>Wrong password!</h1>';
            echo '</div>';
        }

    }
    else
    {
        echo '<div class="welcome-container">';
        echo '<h1>Welcome, user!</h1>';
        echo '</div>';
    }
?>

</body>
</html>
